scripts/benchmark/trainer-benchmark.py

```diff
@@ -0,0 +1,435 @@
+    variation_key = "variation"
+    diff_key = "diff_%"
+
+    sentinel_value = nan
+    if base_variation is not None and len(df[df[variation_key] == base_variation]):
+        # this may still return nan
+        sentinel_value = df.loc[df[variation_key] == base_variation][target_metric_key].item()
+    if math.isnan(sentinel_value):
+        # as a fallback, use the minimal value as the sentinel
+        sentinel_value = df.loc[df[target_metric_key] != nan][target_metric_key].min()
+
+    # create diff column if possible
+    if not math.isnan(sentinel_value):
+        df[diff_key] = df.apply(
+            lambda r: round(100 * (r[target_metric_key] - sentinel_value) / sentinel_value)
+            if not math.isnan(r[target_metric_key])
+            else 0,
+            axis="columns",
+        )
+
+    # re-order columns
+    cols = [variation_key, target_metric_key, diff_key, *report_metric_keys]
+    df = df.reindex(cols, axis="columns")  # reorder cols
+
+    # capitalize
+    df = df.rename(str.capitalize, axis="columns")
+
+    # make the cols as narrow as possible
+    df_github = df.rename(lambda c: c.replace("_", "<br>"), axis="columns")
+    df_console = df.rename(lambda c: c.replace("_", "\n"), axis="columns")
+
+    report = ["", "Copy between the cut-here-lines and paste as is to github or a forum"]
+    report += ["----------8<-----------------8<--------"]
+    report += ["*** Results:", df_github.to_markdown(index=False, floatfmt=".2f")]
+    report += ["```"]
+    report += ["*** Setup:", get_versions()]
+    report += ["*** The benchmark command line was:", get_orig_cmd()]
+    report += ["```"]
+    report += ["----------8<-----------------8<--------"]
+    report += ["*** Results (console):", df_console.to_markdown(index=False, floatfmt=".2f")]
+
+    print("\n\n".join(report))
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--base-cmd",
+        default=None,
+        type=str,
+        required=True,
+        help="Base cmd",
+    )
+    parser.add_argument(
+        "--variations",
+        default=None,
+        type=str,
+        nargs="+",
+        required=True,
+        help="Multi-dimensional variations, example: '|--fp16|--bf16' '|--tf32'",
+    )
+    parser.add_argument(
+        "--base-variation",
+        default=None,
+        type=str,
+        help="Baseline variation to compare to. if None the minimal target value will be used to compare against",
+    )
+    parser.add_argument(
+        "--target-metric-key",
+        default=None,
+        type=str,
+        required=True,
+        help="Target metric key in output_dir/all_results.json, e.g., train_samples_per_second",
+    )
+    parser.add_argument(
+        "--report-metric-keys",
+        default="",
+        type=str,
+        help="Report metric keys - other metric keys from output_dir/all_results.json to report, e.g., train_loss. Use a single argument e.g., 'train_loss train_samples",
```

Comment:
Do you think it could be possible to also provide a link to where additional metric keys could be found?

Reply:
do we have that information? it'd be different in different scripts, no? 

That's why I documented `output_dir/all_results.json` so once you run it you will see all the available keys there for the given type of script.